### PR TITLE
Feature/refactors-services-layout

### PR DESCRIPTION
**Pull request by pcrespov**
## Important refactoring of osparc-sim layout and dockerization:
- services folder:
  - every folder is a service
  - app services can be build, run, composed and deployed using services/docker-compose*.yml files
- Changes to webserver service:
  - client and server under service/web folder
  - client has its own docker-compose for independent development
  - server Dockerfile with doble target (dev and prod)
  - folder structure for python project
- Changes to director service:
  - Dockerfile with double target (dev and prod)
  - folder structure for python project
- .vscode folder becomes a template
- PR #68 and PR #67 already integrated

### CODE REVIEW

**Comment by pcrespov** [comment]
@sanderegg could you please update content of osparc-simcore/services/README.md?

**Comment by sanderegg** [comment]
also I did a PR for the updated README.md, let me know if that worked.

File: Makefile [hunk 1]
```diff
@@ -1,17 +1,19 @@
-# Pipelines Project Make File
 # author: Sylvain Anderegg
 
 PY_FILES = $(strip $(shell find services -iname '*.py'))
 
 build:
 	docker-compose -f services/docker-compose.yml build --no-cache
```
Comment (sanderegg) [inline Makefile]: Why the --no-cache flag in the case of build? for rebuild I agree

**Comment by mguidon** [comment]
@sanderegg As to you second point, we are going to generalize this for generic json data as soon as wee need it. I suggest using a sub folder with `label_name.json` files and a script to attach the content into container labels as we did in the past.

File: .gitignore [hunk 2]
```diff
@@ -107,3 +107,5 @@ node_modules/
 
 # IDEs config
 .vscode/
+
+services/docker-compose.override.yml
```
Comment (sanderegg) [inline .gitignore]: why this? is this generated?

File: Makefile (hunk 1 above)
Comment (sanderegg) [reply Makefile]: I propose we have 3 possible builds:
- dev
- test or production
- deploy
then the following recipies:
- up (calls docker swarm init, docker-compose -f ... up --build)
- down (calls docker-compose -f ... down, docker swarm leave -f)

File: README.md [hunk 3]
```diff
@@ -2,7 +2,20 @@
...
-## Deploy
+## Overview
+
+[![service-web](docs/img/service-web.svg)](http://interactive.blockdiag.com/?compression=deflate&src=eJxdjs0KwjAQhO99imXPFtNbpcYXkR7ys2hw6UqSKiK-uymxiF7nm28Yy-IuPpgTPBsAvJPdOg40ZYRjOpsr6UkyjUOB_tEmirfgKH0YaEDHMnsshX993x5qsEgUyx4bS6x3qu824IQlastz3f4pFtGHSC5LXCXsleqw3ljRUvteGprXG1PtQR0)
+
+
+## Deploy in-place
+
+```bash
+  make build
```
Comment (sanderegg) [inline README.md]: I don't think this is correct as make build, builds the docker-compose.yml and up uses the docker-compose.yml + docker-compose.deploy.yml files

Very nice graphs!

File: Makefile [hunk 4]
```diff
@@ -3,13 +3,16 @@
 PY_FILES = $(strip $(shell find services -iname '*.py'))
 
 build:
+	docker-compose -f services/docker-compose.yml build
```
Comment (sanderegg) [inline Makefile]: My proposal would be 3 flavors:
- dev (for development)
- test or production (for testing in production environment)
- deploy (for the swarm)
then only 2 recipes:
- up (calls docker swarm init, docker-compose ... up --build)
- down (as of now)

File: services/README.md [hunk 5]
```diff
@@ -1,23 +1,55 @@
 # services
 
-Provide access to services from docker registry
+Each folder contains a services that is part or can be spawned by the platform.
+The prefix *dy-* in the naming indicates that this service is not a building
+part of the application (i.e. not listed as an services in the docker-compose file)
+but instead it is *dy*namically spawned by the director as a back-end service.
```
Comment (sanderegg) [inline services/README.md]: computation folder should be then prefixed isn't it?

File: services/web/server/source/interactive_services_manager.py [hunk 6]
```diff
@@ -24,15 +27,20 @@ def retrieve_list_of_services():
     try:
         return director_proxy.retrieve_interactive_services()
     except Exception as err:
+        # FIXME: shouldn't I return a json??
```
Comment (sanderegg) [inline services/web/server/source/interactive_services_manager.py]: is a string not a valid json?

File: services/README.md (hunk 5 above)
Comment (pcrespov) [reply services/README.md]: Computations folder still empty .. I would wait to @mguidon takes the decision

File: .gitignore (hunk 2 above)
Comment (pcrespov) [reply .gitignore]: I use this trick to shorten my docker-compose commands while developing. E.g. if I want to build development images, one would need to call
``` docker-compose -f docker-compose.yml -f docker-compose.development.yml build ```
but instead I copy and rename ``docker-compose.development.yml `` as ''docker-compose.override.yml '' and then I can simplify the call to
``` docker-compose build ```
Check this [doc](https://docs.docker.com/compose/extends/#multiple-compose-files)

**Pull request merged**